### PR TITLE
feat: account history

### PR DESCRIPTION
**Pull request by juceliofloresta**
## Feature Account History #32

```diff
# This feature implements the History of an Account when giving an Account Path.

# Repository and Use Case
+ Get a history of an account

# Value Objects
+ AccountHistory
+ EntryHistory

+ Update RPC SDK with GetAccountHistory method.

- It wasn't implementing the data range filter.

! SDK users must update your client.

@@ Have a great code review xD @@
```

### CODE REVIEW

File: clients/grpc/examples/get_account_history.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	accountOne, err := conn.GetAccountBalance(context.Background(), accountPathOne)
+
+	t = conn.NewTransaction(uuid.New())
+	t.AddEntry(uuid.New(), accountPathOne, accountOne.CurrentVersion(), vos.DebitOperation, 500)
+	t.AddEntry(uuid.New(), accountPathTwo, vos.AnyAccountVersion, vos.CreditOperation, 500)
+	err = conn.SaveTransaction(context.Background(), t)
+	AssertEqual(nil, err)
+
+	accountHistory, err := conn.GetAccountHistory(context.Background(), accountPathOne)
```
Comment (alexgarzao) [inline clients/grpc/examples/get_account_history.go:33]: Assert err == nil after GetAccountHistory to avoid panic during test execution.

Comment (alexgarzao) [reply clients/grpc/examples/get_account_history.go:33]: And test to assure that accountHistory has 2 entries.

File: clients/grpc/examples/get_account_history.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	AssertEqual(1000, accountHistory[0].Amount())
+	AssertEqual(vos.CreditOperation, accountHistory[0].Operation())
+
+	AssertEqual(500, accountHistory[1].Amount())
+	AssertEqual(vos.DebitOperation, accountHistory[1].Operation())
+
```
Comment (alexgarzao) [inline clients/grpc/examples/get_account_history.go:40]: I think that is a good idea to check for accountPathTwo history too

File: clients/grpc/examples/get_account_history.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	accountOne, err = conn.GetAccountBalance(context.Background(), accountPathOne)
+
+	t = conn.NewTransaction(uuid.New())
+	t.AddEntry(uuid.New(), accountPathOne, accountOne.CurrentVersion(), vos.DebitOperation, 1000)
+	t.AddEntry(uuid.New(), accountPathTwo, vos.AnyAccountVersion, vos.CreditOperation, 1000)
+	err = conn.SaveTransaction(context.Background(), t)
+	AssertEqual(nil, err)
+
+	accountHistory, err := conn.GetAccountHistory(context.Background(), accountPathOne)
```
Comment (alexgarzao) [inline clients/grpc/examples/get_account_history.go:81]: Check err and accountHistory size here to avoid panic

**Review by alexgarzao — CHANGES_REQUESTED** [review]
Very nice PR!

We must use NewRelic, when necessary. And there is a possible panic during tests.

File: clients/grpc/examples/get_account_history.go (hunk 1 above)
Comment (juceliofloresta) [reply clients/grpc/examples/get_account_history.go:33]: :+1: :heavy_check_mark:

File: clients/grpc/examples/get_account_history.go (hunk 3 above)
Comment (juceliofloresta) [reply clients/grpc/examples/get_account_history.go:81]: :+1: :heavy_check_mark:

File: clients/grpc/examples/get_account_history.go (hunk 2 above)
Comment (juceliofloresta) [reply clients/grpc/examples/get_account_history.go:40]: :+1: :heavy_check_mark:

**Review by alexgarzao — APPROVED** [review]

**Review by alexmourapb — APPROVED** [review]

**Pull request merged**